core/src/test/java/org/elasticsearch/action/bulk/TransportShardBulkActionTests.java

```diff
@@ -503,6 +506,53 @@ public void testCalculateTranslogLocation() throws Exception {
 
     }
 
+    public void testMappingUpdateDoesNotParseDocTwice() throws Exception {
+        IndexMetaData metaData = indexMetaData();
+        logger.info("--> metadata.getIndex(): {}", metaData.getIndex());
+        final IndexShard shard = newStartedShard(true);
+
+        IndexRequest request = new IndexRequest("index", "type", "id")
+                .source(Requests.INDEX_CONTENT_TYPE, "foo", "bar");
+
+        TransportShardBulkAction.executeIndexRequestOnPrimary(request, shard,
```

Comment:
This is a bit confusing to me given the name of the test - in this case we would parse the doc twice, and that's OK?